sdk/python/feast/infra/provider.py

```diff
@@ -0,0 +1,49 @@
+import abc
+from typing import List
+
+from feast import FeatureTable
+from feast.repo_config import RepoConfig
+
+
+class Provider(abc.ABC):
+    @abc.abstractmethod
+    def update_infra(
+        self,
+        project: str,
```

Comment:
@jklegar @oavdeev @jparthasarthy can we maybe make a decision on something. Is `project` === `repo`? If not, what is the relationship.

Reply:
Roughly yes, the idea is that you can set project field to something in the repo config. That way all the infra created by `feast deploy` will be namespaced† by the project name. So you'd be able, for example, to `feast deploy` multiple independent feature repos to the same AWS account and they won't interfere with each other.

† exact meaning of "namespaced" is provider-dependent, could be a resource name prefix or some kind of tag

Reply:
I agree with this approach, but does this imply that we should remove projects from our API itself? Because we would never by able to have an operation like `list_projects()` that returns more than one project right?

Reply:
I think so, yes